vars/signArtifacts.groovy

```diff
@@ -0,0 +1,25 @@
+void call(Map args = [:]) {
+
+    environment {
+        // These ENV variables are required by https://github.com/opensearch-project/opensearch-signer-client
+        // This client is invoked internally by the sign script.
+        ROLE = "${SIGNER_CLIENT_ROLE}"
+        EXTERNAL_ID = "${SIGNER_CLIENT_EXTERNAL_ID}"
+        UNSIGNED_BUCKET = "${SIGNER_CLIENT_UNSIGNED_BUCKET}"
+        SIGNED_BUCKET = "${SIGNER_CLIENT_SIGNED_BUCKET}"
+    }
+
+    git url: 'https://github.com/opensearch-project/opensearch-build.git', branch: 'main'
+
+    sh('curl https://artifacts.opensearch.org/publickeys/opensearch.pgp -o $WORKSPACE/opensearch.pgp')
+    sh('gpg --import $WORKSPACE/opensearch.pgp')
+
+    def signatureType = ".sig"
```

Comment:
The value should come from args.signatureType not hard coded (same for other parameters)

See https://github.com/opensearch-project/opensearch-build/blob/50552294f6830d7412d66831798fe15acf89e43d/vars/buildManifest.groovy for an example.